spec/lrama/warnings/required_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+RSpec.describe Lrama::Warnings::Required do
+  describe "#warn" do
+    let(:y) do
+      <<~STR
+        %require "3.0"
+        %{
+        // Prologue
+        %}
+        %union {
+            int i;
+        }
+        %token <i> tNUMBER
+        %%
+        program: tNUMBER
+                ;
+      STR
+    end
+
+    context "when warnings true" do
+      it "has warns for require is provided for compatibility with bison" do
+        grammar = Lrama::Parser.new(y, "states/required.y").parse
+        grammar.prepare
+        grammar.validate!
+        states = Lrama::States.new(grammar, Lrama::Tracer.new(Lrama::Logger.new))
+        states.compute
+        logger = Lrama::Logger.new
+        allow(logger).to receive(:warn)
+        Lrama::Warnings.new(logger, true).warn(grammar, states)
+        expect(logger).to have_received(:warn).with("currently, %require is simply valid as a grammar but does nothing")
+      end
+    end
+
+    context "when warnings false" do
+      it "has not warns for require is provided for compatibility with bison" do
+        grammar = Lrama::Parser.new(y, "states/required.y").parse
+        grammar.prepare
+        grammar.validate!
+        states = Lrama::States.new(grammar, Lrama::Tracer.new(Lrama::Logger.new))
+        states.compute
+        logger = Lrama::Logger.new
+        allow(logger).to receive(:warn)
+        Lrama::Warnings.new(logger, false).warn(grammar, states)
+        expect(logger).not_to have_received(:warn).with("currently, %require is simply valid as a grammar but does nothing")
```

Comment:
It's better to check `logger#warn` is not called regardless its message otherwise the test case will pass when different message is send.

Reply:
Indeed. Thank you!